app-cpp/app355.h

```diff
@@ -6,6 +6,142 @@
 #define _snprintf_s (snprintf)
 #endif // WIN32
 
+// std
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <list>
+#include <mutex>
+#include <string>
+#include <tuple>
+#include <thread>
+#include <vector>
+
+// self
+#include "libsat355.h"
+
+#pragma region class OrbitalData
+class OrbitalData
+{
+private:
+    sat355::TLE mTLE;
+    double mLatitude;
+    double mLongitude;
+    double mAltitude;
+    std::string mName;
+    double mMeanMotion;
+
+public:
+    OrbitalData(sat355::TLE inTLE, double latitude, double longitude, double altitude)
+        : mTLE{std::move(inTLE)}, mLatitude(latitude), mLongitude(longitude), mAltitude(altitude)
+    {
+        mName = mTLE.GetName();
+        mMeanMotion = mTLE.GetMeanMotion();
+    }
+
+    OrbitalData() = delete;
+
+    ~OrbitalData() = default;
+
+    const sat355::TLE& GetTLE() const 
+    { 
+        return mTLE; 
+    }
+
+    double GetLatitude() const 
+    { 
+        return mLatitude; 
+    }
+    void SetLatitude(double latitude) 
+    { 
+        mLatitude = latitude; 
+    }
+
+    double GetLongitude() const 
+    { 
+        return mLongitude; 
+    }
+    void SetLongitude(double longitude) 
+    { 
+        mLongitude = longitude; 
+    }
+
+    double GetAltitude() const 
+    { 
+        return mAltitude; 
+    }
+    void SetAltitude(double altitude) 
+    { 
+        mAltitude = altitude; 
+    }
+};
+#pragma endregion {}
+
+//--------------------------------------------------
+#pragma region class SatOrbit
+// abstract base class
+
+/// @brief SatOrbit acts as the API for calculating then sorting satellite orbits into groups or, trains"
+class SatOrbit
+{
+public:
+    enum class SatOrbitKind
+    {
+        kDefault = 0,
+        kSingle,
+        kMulti
+    };
+
+ // Interface
+public:
+    /// @brief SatOrbit is purely abstract, therefore use SatOrbit::Make() instead to create SatOrbit objects
+    SatOrbit() = default;
+
+    /// @brief dtor is default, giving access to RO5 methods
+    virtual ~SatOrbit() = default;
```

Comment:
Place the ~dtor() at the end of the `public` section, rather than the top.
you should reserve the top of the `public` section for the most important methods...
for instance: `Make()`